### ISSUE TITLE
Iteration performance

### ISSUE THREAD

**Issue by 9il** [libmir/dcv#27]
`aSlice[i]` requires 1 addition and 1 multiplication
`aSlice[i, j]` requires 2 additions and 2 multiplications
90% of algorithm can be iterated using ndslice.algorithm (ndMap, ndReduce). Other 10% can use `front!d`, `popFront!d` methods

**Comment by 9il** [comment]
For example, this code can be optimizaed using `assumeSameStructure` (zip analog) and `ndEach`

``` d
        foreach(r; iota(f1.height).parallel)
        {
            foreach(c; 0 .. f1.width)
            {
                auto id = r * f1.width + c;
                f1s[r, c] = cast(float)f1d[id];
                f2s[r, c] = cast(float)f2d[id];
            }
        }
```

As Walter said, idiomatic D code should not have loops. This is really true for ndslice.

**Comment by ljubobratovicrelja** [comment]
I agree with everything you've stated. I assume we could include these changes in same basket as migration to Mir and ndslice.algorithm?

**Comment by 9il** [comment]
Yes

**Comment by 9il** [comment]
The code like `[i, j]` still exist, for example in https://github.com/ljubobratovicrelja/dcv/blob/master/source/dcv/tracking/opticalflow/lucaskanade.d

**Comment by ljubobratovicrelja** [comment]
Aw, man... :| Yea, I haven't cleaned optical flow modules, wanted to cut #58 short. Reopening...
